### PR TITLE
UI Enhancements

### PR DESCRIPTION
**Pull request by martyrion**
Please **forgive me** in advance. I made a mess with a lot of unnecessary commits and reverts.

Changes:

- og.css : refactored for reusable styles and codes
- Layer Switcher Control: added drag&drop for ZIndex change. Made sure layers added later via js still update correctly, changes in styling. Dialog closes when you click outside
- GeoImageDragger: changed cursor at corners, made ON/OFF button
- Ruler: made on off button

### CODE REVIEW

File: css/og.css [hunk 1]
```diff
@@ -129,93 +128,170 @@
...
 
 /* =============================== */
-/* LAYER SWITCHER */
-.ogLayerSwitcherButton {
-    /* background-clip: padding-box; */
-    height: 40px;
+/*         MENU V-BAR STYLING      */
+/* =============================== */
+
+.menu-bar-vertical{
```
Comment (Zemledelec) [inline css/og.css]: Please, add prefix **og-**: og-menu..

File: examples/userInterface/userInteface.html [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                url: "https://api.mapbox.com/styles/v1/mapbox/satellite-v9/tiles/256/{z}/{x}/{y}?access_token=pk.eyJ1IjoibWdldmxpY2hzY2FuZXgiLCJhIjoiY2pwcGdsaXlnMDQzdDQybXhsOWZlbXBvdSJ9.fR2YE-ehJA4iajaJBAPKvw",
+                visibility: false,
+                attribution: `@2014 MapQuest - Portions @2014 "Map data @
+            <a target="_blank" href="//www.openstreetmap.org/">OpenStreetMap</a> and contributors,
+            <a target="_blank" href="//opendatacommons.org/licenses/odbl/"> CC-BY-SA</a>"`
+            });
+
+            var sat2 = new XYZ("MapQuest Sat Labels", {
+                fading: true,
+                shininess: 20,
```
Comment (Zemledelec) [inline examples/userInterface/userInteface.html:58]: Btw, shininess, specular, diffuse, and ambient parameters do not exist anymore. It was a long time ago.

File: src/og/control/LayerSwitcher.js [hunk 3]
```diff
@@ -5,20 +5,24 @@
...
  * @param {Object} [options] - Control options.
  */
 class LayerSwitcher extends Control {
     constructor(options) {
         super(options);
-
         this.dialog = null;
-        this.baseLayersDiv = null;
-        this.overlaysDiv = null;
+        this.baseLayersContainer = null;
```
Comment (Zemledelec) [inline src/og/control/LayerSwitcher.js]: Maybe for future things, I suggest using prefix $ for dom variables, like this.$baseLayerContainer or even $layers. It's not needed to change now, but if you find it good, we can use this rule.

File: src/og/control/UIhelpers.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+'use strict';
```
Comment (Zemledelec) [inline src/og/control/UIhelpers.js:1]: This is cool!:)

File: src/og/control/ruler/Ruler.js [hunk 5]
```diff
@@ -38,6 +45,26 @@ class Ruler extends Control {
     ondeactivate() {
         this.renderer.removeNode(this._rulerScene);
     }
-}
+
+
+    createMenuBtn() {
+        let btn = elementFactory('div', { id: 'ruler-menu-btn', class: 'ruler menu-btn OFF' },
```
Comment (Zemledelec) [inline src/og/control/ruler/Ruler.js]: need prefix **og-**

File: src/og/control/ruler/RulerScene.js [hunk 6]
```diff
@@ -112,12 +112,14 @@ class RulerScene extends RenderNode {
             entities: [this._trackEntity, this._propsLabel],
             pickingEnabled: false,
             polygonOffsetUnits: -1.0,
-            relativeToGround: true
+            relativeToGround: true,
+            displayInLayerSwitcher: false
```
Comment (Zemledelec) [inline src/og/control/ruler/RulerScene.js:116]: Nice!

Comment (Zemledelec) [reply src/og/control/ruler/RulerScene.js:116]: Also `displayInLayerSwitcher` must be in the `Layer `class constructor.

**Review by Zemledelec — CHANGES_REQUESTED** [review]
Super! Just a few improves needed, I've made comments!
FYI @pavletto if you have something please respond.

File: examples/userInterface/userInteface.html (hunk 2 above)
Comment (martyrion) [reply examples/userInterface/userInteface.html:58]: Done.

File: src/og/control/LayerSwitcher.js (hunk 3 above)
Comment (martyrion) [reply src/og/control/LayerSwitcher.js]: Maybe _**this.DOM_baselayersContainer**_ would be better?

File: css/og.css (hunk 1 above)
Comment (martyrion) [reply css/og.css]: Done

File: src/og/control/ruler/Ruler.js (hunk 5 above)
Comment (martyrion) [reply src/og/control/ruler/Ruler.js]: Done. you want the prefix og- to be present in all css classes and ids? In my opinion it reduces a bit the readability of the code.

I mean we are in the src/ of Openglobus after all

File: src/og/control/ruler/RulerScene.js (hunk 6 above)
Comment (martyrion) [reply src/og/control/ruler/RulerScene.js:116]: I think it is already. I added it to the  @params of Layer.js

File: css/og.css [hunk 7]
```diff
@@ -129,93 +128,170 @@
...
+}
+
+
+/* =============================== */
+/*         MENU ΒΤΝ STYLING        */
+/*         General Settings        */
+/* =============================== */
+
+/* All buttons have this class + OFF if they are OFF */
+.menu-btn{
```
Comment (Zemledelec) [inline css/og.css]: Sorry, I had to say that adding **og-** touches every style name we use in native controls. Please, add **og-** for all new styles.

File: css/og.css [hunk 8]
```diff
@@ -129,93 +128,170 @@
...
 
+.layer-switcher.dialog {
```
Comment (Zemledelec) [inline css/og.css]: og-

File: css/og.css [hunk 9]
```diff
@@ -129,93 +128,170 @@
...
 
+.layer-switcher.dialog { 
+    top: 32px; 
 }
 
-.layersEntry {
-   display: flex;
-   align-items: center;
-   height: 24px;
+.layer-container{
```
Comment (Zemledelec) [inline css/og.css]: og- etc.

File: css/og.css [hunk 10]
```diff
@@ -129,93 +128,170 @@
...
+    /* opacity: 0.5; */
 }
+
+
 /* =============================== */
+/* GEO IMAGE DRAGGER */
+
+/* Just set position and background url */
+
+#geo-image-dragger-menu-btn{
```
Comment (Zemledelec) [inline css/og.css]: og-

**Review by Zemledelec — CHANGES_REQUESTED** [review]
Forget to mention that **og-** for all styles for native controls needed.

**Comment by martyrion** [comment]
Sorry didn't see your new issue before. I have a solution committed pls let me know.

**Review by Zemledelec — APPROVED** [review]
Good!

**Pull request merged**